### PR TITLE
[ENG-597] Adds a Jira client allowing to post comments to Jira issues

### PR DESCRIPTION
**Pull request by 0mega**
This PR creates a wrapper for jira-ruby gem, and allows to post comments on behalf of a user configured through environment variables.

### CODE REVIEW

File: lib/clients/jira.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class JiraClient
+  def self.post_comment(issue_id, comment_msg)
+    issue = get_issue(issue_id)
+    comment = issue.comments.build
+    comment.save('body': comment_msg)
+  end
+
+  def self.get_issue(id)
+    client = jira_client
```
Comment (dideler) [inline lib/clients/jira.rb]: nitpick: this assignment doesn't seem necessary

File: lib/clients/jira.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  def self.jira_client
+    options = {
+      username: ENV['JIRA_USER'],
+      password: ENV['JIRA_PASSWD'],
+      site: ENV['JIRA_HOST'],
+      context_path: ENV['JIRA_PATH'],
+      auth_type: :basic,
+      read_timeout: 120,
+    }
```
Comment (dideler) [inline lib/clients/jira.rb]: minor: options can also be cached to avoid recreating every time the client is used

File: lib/clients/jira.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    client.Issue.find(id)
+  end
+  private_class_method :get_issue
+
+  def self.jira_client
+    options = {
+      username: ENV['JIRA_USER'],
+      password: ENV['JIRA_PASSWD'],
+      site: ENV['JIRA_HOST'],
+      context_path: ENV['JIRA_PATH'],
```
Comment (dideler) [inline lib/clients/jira.rb]: nitpick: do you think it's worth setting these as constants in `application.rb`?

**Comment by dideler** [comment]
:+1: 

We should add tests at some point, but not a blocker now since the class isn't being used yet.

File: config/environments/development.rb [hunk 4]
```diff
@@ -35,6 +35,4 @@
   # config.action_view.raise_on_missing_translations = true
 
   config.active_job.queue_adapter = :delayed_job
-
-  config.jira_host_name = ENV.fetch('JIRA_HOST_NAME', 'localhost')
```
Comment (dideler) [inline config/environments/development.rb]: Since this had a fallback, we should ensure we have the value in our `.env.development`

Comment (0mega) [inline config/environments/development.rb]: Agreed, although it's very unlikely you will be running Jira on localhost.

**Comment by andistuder** [comment]
LGTM

**Pull request merged**